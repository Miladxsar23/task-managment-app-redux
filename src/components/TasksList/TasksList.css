.tasks-list-item {
  min-height: 100vh;
}
/*# sourceMappingURL=TasksList.css.map */