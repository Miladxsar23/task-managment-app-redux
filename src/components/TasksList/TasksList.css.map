{
    "version": 3,
    "mappings": "AAGG,AAAD,gBAAM,CAAC;EACL,UAAU,EAAE,KAAK;CAClB",
    "sources": [
        "TasksList.scss"
    ],
    "names": [],
    "file": "TasksList.css"
}