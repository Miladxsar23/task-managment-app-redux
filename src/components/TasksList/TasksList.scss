.tasks-list {
  &-tltle {
  }
  &-item {
    min-height: 100vh;
  }
}
