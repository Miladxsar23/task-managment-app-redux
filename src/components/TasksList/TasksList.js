import * as React from "react";
import "./TasksList.scss";
import Task from "../Task/Task";
const TasksList = ({ filteredTasks, onEditTask, title }) => {
  const rows = filteredTasks.map((task) => {
    return <Task key={task.id} task={task} onEditTask={onEditTask} />;
  });
  return (
    <div className="tasks col-sm-12 col-md-3">
      <div className="tasks-list">
        <div className="tasks-list-title">
          <h4>{title}</h4>
        </div>
        <hr />
        <div className="tasks-lists-item d-flex flex-column justify-content-start align-items-center my-2">
          {rows}
        </div>
      </div>
    </div>
  );
};

export default TasksList;
