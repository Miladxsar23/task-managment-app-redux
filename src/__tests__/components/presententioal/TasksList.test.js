import * as React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TasksList from "../../../components/TasksList/TasksList";
describe("TasksList", () => {
  test("should render a status", () => {
    render(
      <TasksList filteredTasks={[]} onEditTask={() => {}} title="In Progress" />
    );
    expect(screen.getByText(/In Progress/)).toBeInTheDocument();
  });
  test("snapshot test without tasks", () => {
    const tree = renderer.create(
      <TasksList filteredTasks={[]} onEditTask={() => {}} title="In Progress" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render a Task component for each task", () => {
    const mockTasks = [
      {
        id: 1,
        title: "A",
        description: "a",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
      {
        id: 2,
        title: "B",
        description: "b",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
      {
        id: 3,
        title: "C",
        description: "c",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
    ];
    render(
      <TasksList
        filteredTasks={mockTasks}
        onEditTask={() => {}}
        title="Unstarted"
      />
    );
    expect(
      screen.getAllByRole("combobox", { label: "Change status of task" })
    ).toHaveLength(3);
  });
  test("snapshot test with tasks", () => {
    const mockTasks = [
      {
        id: 1,
        title: "A",
        description: "a",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
      {
        id: 2,
        title: "B",
        description: "b",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
      {
        id: 3,
        title: "C",
        description: "c",
        status: "Unstarted",
        timer: 0,
        projectId: 1,
      },
    ];
    const tree = renderer.create(
      <TasksList
        filteredTasks={mockTasks}
        onEditTask={() => {}}
        title="Unstarted"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
