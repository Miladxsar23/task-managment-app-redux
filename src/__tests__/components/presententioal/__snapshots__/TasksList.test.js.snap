// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksList snapshot test with tasks 1`] = `
<div
  className="tasks col-sm-12 col-md-3"
>
  <div
    className="tasks-list"
  >
    <div
      className="tasks-list-title"
    >
      <h4>
        Unstarted
      </h4>
    </div>
    <hr />
    <div
      className="tasks-lists-item d-flex flex-column justify-content-start align-items-center my-2"
    >
      <div
        className="task w-100 d-flex flex-column p-3 my-2 rounded"
      >
        <div
          className="task-header d-flex flex-row justify-content-between align-items-center"
        >
          <div
            className="task-header-title text-primary"
          >
            <strong
              className="title"
            >
              A
            </strong>
          </div>
          <div
            className="task-header-status"
          >
            <select
              aria-label="Change status of task"
              className="form-select form-select-sm"
              onChange={[Function]}
              value="Unstarted"
            >
              <option
                value="Unstarted"
              >
                Unstarted
              </option>
              <option
                value="In Progress"
              >
                In Progress
              </option>
              <option
                value="Completed"
              >
                Completed
              </option>
            </select>
          </div>
        </div>
        <hr />
        <div
          className="task-body"
        >
          a
        </div>
        <div
          className="task-footer"
        >
          <div
            className="Timer"
          >
            <small>
              0
              s
            </small>
          </div>
        </div>
      </div>
      <div
        className="task w-100 d-flex flex-column p-3 my-2 rounded"
      >
        <div
          className="task-header d-flex flex-row justify-content-between align-items-center"
        >
          <div
            className="task-header-title text-primary"
          >
            <strong
              className="title"
            >
              B
            </strong>
          </div>
          <div
            className="task-header-status"
          >
            <select
              aria-label="Change status of task"
              className="form-select form-select-sm"
              onChange={[Function]}
              value="Unstarted"
            >
              <option
                value="Unstarted"
              >
                Unstarted
              </option>
              <option
                value="In Progress"
              >
                In Progress
              </option>
              <option
                value="Completed"
              >
                Completed
              </option>
            </select>
          </div>
        </div>
        <hr />
        <div
          className="task-body"
        >
          b
        </div>
        <div
          className="task-footer"
        >
          <div
            className="Timer"
          >
            <small>
              0
              s
            </small>
          </div>
        </div>
      </div>
      <div
        className="task w-100 d-flex flex-column p-3 my-2 rounded"
      >
        <div
          className="task-header d-flex flex-row justify-content-between align-items-center"
        >
          <div
            className="task-header-title text-primary"
          >
            <strong
              className="title"
            >
              C
            </strong>
          </div>
          <div
            className="task-header-status"
          >
            <select
              aria-label="Change status of task"
              className="form-select form-select-sm"
              onChange={[Function]}
              value="Unstarted"
            >
              <option
                value="Unstarted"
              >
                Unstarted
              </option>
              <option
                value="In Progress"
              >
                In Progress
              </option>
              <option
                value="Completed"
              >
                Completed
              </option>
            </select>
          </div>
        </div>
        <hr />
        <div
          className="task-body"
        >
          c
        </div>
        <div
          className="task-footer"
        >
          <div
            className="Timer"
          >
            <small>
              0
              s
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TasksList snapshot test without tasks 1`] = `
<div
  className="tasks col-sm-12 col-md-3"
>
  <div
    className="tasks-list"
  >
    <div
      className="tasks-list-title"
    >
      <h4>
        In Progress
      </h4>
    </div>
    <hr />
    <div
      className="tasks-lists-item d-flex flex-column justify-content-start align-items-center my-2"
    />
  </div>
</div>
`;
